toolchain/lower/function_context.cpp

```diff
@@ -37,19 +37,33 @@ auto FunctionContext::TryToReuseBlock(SemIR::InstBlockId block_id,
   return true;
 }
 
+// Structure for the core handler dispatch.
+using DispatchFunctionT = auto(FunctionContext& context, SemIR::InstId inst_id,
+                               SemIR::Inst inst) -> void;
+using DispatchTableT =
+    std::array<DispatchFunctionT*, SemIR::InstKind::EnumCount>;
+
+// Transforms a parse node ID to a typed parse node, which is then passed to the
+// handler.
+#define CARBON_SEM_IR_INST_KIND(Name)                                  \
+  auto Dispatch##Name(FunctionContext& context, SemIR::InstId inst_id, \
```

Comment:
If you don't switch to using a lambda as I suggest in the other comment, this should be declared `static`.

Reply:
Done